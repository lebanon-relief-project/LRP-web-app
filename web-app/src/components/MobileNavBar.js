import React, { useRef } from "react";
import styled from "styled-components";
import logo from "../assets/images/logo.svg";
import devices from "../styles/Devices";
import Burger from "./Burger";
import { Routes } from "../App.js";
import { NavLink } from "react-router-dom";

const MobileNavbar = ({ open, setOpen }) => {
  const node = useRef();
  return (
    <StyledNav open={open}>
      <LogoLink exact to={Routes.HOME}>
        <LogoImage src={logo} alt="Logo" />
      </LogoLink>
      <BurgerWrapper ref={node}>
        <Burger open={open} setOpen={setOpen} />
      </BurgerWrapper>
    </StyledNav>
  );
};

const BurgerWrapper = styled.nav`
  display: flex;
  flex: 1;
  align-items: center;
`;

const StyledNav = styled.nav`
  position: fixed;
  display: flex;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
  background-color: white;
  @media (min-width: ${devices.mobile}) {
    display: none;
  }

  height: 64px;
  max-height: 64px;
  width: 100%;
  z-index: 100;
`;

const LogoImage = styled.img`
  max-height: 80%;
`;

const LogoLink = styled(NavLink)`
  min-width: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export default MobileNavbar;
