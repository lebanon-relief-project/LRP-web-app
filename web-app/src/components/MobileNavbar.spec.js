import React from "react";
import { render } from "@testing-library/react";
import MobileNavBar from "./MobileNavBar";
import "jest-styled-components";

jest.mock("react-router-dom", () => ({
  Link: ({ to }) => <div data-testid={`link_${to}`}></div>,
  NavLink: ({ to }) => <div data-testid={`navlink_${to}`}></div>,
}));

describe("the MobileNavBar component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderMobileNavBar());

    expect(container).toMatchSnapshot();
  });
});

const renderMobileNavBar = () => {
  return render(<MobileNavBar />);
};
