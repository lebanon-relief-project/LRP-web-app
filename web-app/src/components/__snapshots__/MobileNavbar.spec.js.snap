// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MobileNavBar component should match the snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  position: fixed;
  right: 2rem;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  outline: none;
}

.c2 span {
  width: 2rem;
  height: 0.25rem;
  background: #002766;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c2 span:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c2 span:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c2 span:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  background-color: white;
  height: 64px;
  max-height: 64px;
  width: 100%;
  z-index: 100;
}

@media (min-width:576px) {
  .c0 {
    display: none;
  }
}

<div>
  <nav
    class="c0"
  >
    <div
      data-testid="navlink_/"
    />
    <nav
      class="c1"
    >
      <button
        aria-expanded="false"
        aria-label="Toggle menu"
        class="c2"
        data-testid="burgerButton"
      >
        <span />
        <span />
        <span />
      </button>
    </nav>
  </nav>
</div>
`;
